os/kernel/log_dump/log_dump.c

```diff
@@ -613,3 +635,134 @@ int log_dump(int argc, char *argv[])
 		}
 	}
 }
+
+
+void assert_log_init(){
+	assert_situation=0;
+	assert_log_index=0;
+}
+int assert_log_save(char ch){
+	if(assert_log_index<ASSERT_LOG_BUF_SIZE) assert_log[assert_log_index++]=ch;
+}
+int get_assert_log_save_size(){
+	return log_len;
+}
+int set_assert_situation(){
+	lldbg("\n\n\n===============================FROM THIS, lldbg LOG WILL BE SAVED!!================================\n\n\n\n");
+	assert_situation=1;
+}
+int set_notassert_situation(){
+	lldbg("\n\n\n=============================================TO THIS===============================================\n\n\n");
+	
+	assert_situation=0;
+}
+int get_assert_log(char *buf,int size){
+	int i;
+	for(i=0;i<get_assert_log_save_size;i++){
+		if(i>=size) break;
+		buf[i]=assert_log[i];
+	}
+	return i;
+}
+void append_assert_log(const char *fmt, va_list ap){
+	if (log_len>=ASSERT_LOG_BUF_SIZE - 1){
+		return ;
+	}
+	size_t remain=ASSERT_LOG_BUF_SIZE - log_len-1;
+	int written = vsnprintf(assert_log + log_len, remain+1,fmt,ap);
+	if(written<0) return;
+	if((size_t)written > remain){
+		log_len = ASSERT_LOG_BUF_SIZE -1 ;
+	}
+	else{
+		log_len +=(size_t)written;
+	}
+	assert_log[log_len]='\0';
+}
+void assert_log_to_buffer(FAR const char *fmt, va_list ap){
+	if(assert_situation){
+		va_list ap2;
+		va_copy(ap2,ap);
+		append_assert_log(fmt,ap2);
+		va_end(ap2);
+	}
+}
+
+
+
+int assert_log_to_file(){
+	lldbg("\n\n\n===============assert log file saving start==================\n\n\n");
+	lldbg("\nlog_save_size is %d\n",get_assert_log_save_size());
+	int fd=open("/mnt/assert_logsave.txt",O_WRONLY|O_CREAT|O_TRUNC);
+	if(fd<0){
+		lldbg("file open failed\n");
+		return -1;
+	}
+	lldbg("file open success\n");
+	int log_size=write(fd,assert_log,strlen(assert_log));
+	close(fd);
+	if(log_size<0){
+		lldbg("file write failed\n");
+		return -1;
+	}
+	lldbg("file write success, count:%d\n",log_size);
+	lldbg("\n\n\n===============assert log file saving end=====================\n\n\n");
+	return log_size;
+}
+void assert_log_compress(){
+
+	lldbg("\n\n================compress start==============\n\n");
+	compressed_buf_size=ASSERT_LOG_BUF_SIZE;
+	compress_block(compressed_buf,&compressed_buf_size,assert_log,ASSERT_LOG_BUF_SIZE);
+	lldbg("compressed buf size: %d\n",compressed_buf_size);
+	lldbg("\n====================compress end =====================\n");
+		
+}
```

Comment:
The code is exculed during crash recovery.

So, we need check time for compress.
If It is long, not compress is better.

for it, How about make configuration compress or not.